moses/moses/main/populate_atomspace.cc

```diff
@@ -0,0 +1,70 @@
+#include <moses/data/table/table.h>
+#include <moses/data/table/table_io.h>
+#include <opencog/atomspace/AtomSpace.h>
+#include <moses/utils/value_key.h>
+#include <opencog/atoms/base/Link.h>
+#include "populate_atomspace.h"
+
+namespace opencog {
+    namespace moses {
+
+        void populate(AtomSpace *as, ITable &itable) {
+
+            std::vector<multi_type_seq>::const_iterator it, end;
+            int col_size = itable.get_types().size();
+            for (int i = 0; i < col_size; i++) {
+                id::type_node col_type = itable.get_types().at(i);
+
+                int row_size = itable.size();
+
+                switch (col_type) {
+                    case id::boolean_type: {
+                        std::vector<ProtoAtomPtr> col_values = {};
+                        Handle feature = createNode(SCHEMA_NODE,"f"+ std::to_string(i+1));
```

Comment:
Actually the label corresponding to `i` should be used, not `"f<i+1>"`, see `ITable::get_labels()`.